app/lib/filters/nested_filter.rb

```diff
@@ -0,0 +1,28 @@
+module Filters
+  class NestedFilter < Filter
+    def initialize(facet, params)
+      @main_facet_key = params["main_facet_key"]
+      @sub_facet_key = params["sub_facet_key"]
+      @main_facet_value = params[@main_facet_key]
+      @sub_facet_value = params[@sub_facet_key]
+
+      super
+    end
+
+    def query_hash
+      return {} unless @main_facet_value && @sub_facet_value
+
+      { @main_facet_key => [@main_facet_value], @sub_facet_key => [@sub_facet_value] }
```

Comment:
This would probably be more readable spread over multiple lines. (I'd be happy with a one-liner if these weren't array values)